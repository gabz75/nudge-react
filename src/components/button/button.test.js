import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';

import Button from './button';

it('renders', () => {
  const { asFragment } = render(<Button />);
  expect(asFragment()).toMatchSnapshot();
});

describe('prop `label`', () => {
  it('renders with label="Submit', () => {
    const { asFragment } = render(<Button label="Submit" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with label="Ok', () => {
    const { asFragment } = render(<Button label="Ok" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('prop `type`', () => {
  it('renders with type="submit"', () => {
    const { asFragment } = render(<Button type="submit" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with type="button"', () => {
    const { asFragment } = render(<Button type="button" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('prop `className`', () => {
  it('renders with className="foobar"', () => {
    const { getByTestId } = render(<Button className="foobar" />);
    expect(getByTestId('button')).toHaveClass('foobar');
  });
});

describe('prop `size`', () => {
  it('console.error and throws given an invalid size', () => {
    /* eslint-disable no-console */
    const defaultConsoleError = console.error;
    console.error = jest.fn();
    expect(() => render(<Button size="foobar" />)).toThrow(new Error('Arguments after key must be arrays'));
    expect(console.error).toHaveBeenCalled();
    console.error = defaultConsoleError;
    /* eslint-enable no-console */
  });

  it('renders for `sm`', () => {
    const { asFragment } = render(<Button size="sm" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders for `md`', () => {
    const { asFragment } = render(<Button size="md" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('prop `onClick`', () => {
  it('executes a callback', () => {
    const onClickHandler = jest.fn();
    const { getByTestId } = render(<Button onClick={onClickHandler} />);
    getByTestId('button').click();
    expect(onClickHandler).toHaveBeenCalled();
  });
});
