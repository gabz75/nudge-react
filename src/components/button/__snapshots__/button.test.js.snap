// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prop \`label\` renders with label="Ok 1`] = `
<DocumentFragment>
  <button
    class="bg-cta hover:bg-blue-400 text-white border-b-4 border-yellow-1 hover:border-blue-500 rounded py-2 px-4 text-sm font-bold"
    data-testid="button"
    type="button"
  >
    Ok
  </button>
</DocumentFragment>
`;

exports[`prop \`label\` renders with label="Submit 1`] = `
<DocumentFragment>
  <button
    class="bg-cta hover:bg-blue-400 text-white border-b-4 border-yellow-1 hover:border-blue-500 rounded py-2 px-4 text-sm font-bold"
    data-testid="button"
    type="button"
  >
    Submit
  </button>
</DocumentFragment>
`;

exports[`prop \`size\` renders for \`md\` 1`] = `
<DocumentFragment>
  <button
    class="bg-cta hover:bg-blue-400 text-white border-b-4 border-yellow-1 hover:border-blue-500 rounded py-2 px-4 text-sm font-bold"
    data-testid="button"
    type="button"
  />
</DocumentFragment>
`;

exports[`prop \`size\` renders for \`sm\` 1`] = `
<DocumentFragment>
  <button
    class="bg-cta hover:bg-blue-400 text-white border-b-4 border-yellow-1 hover:border-blue-500 rounded py-1 px-2 text-xs font-bold"
    data-testid="button"
    type="button"
  />
</DocumentFragment>
`;

exports[`prop \`type\` renders with type="button" 1`] = `
<DocumentFragment>
  <button
    class="bg-cta hover:bg-blue-400 text-white border-b-4 border-yellow-1 hover:border-blue-500 rounded py-2 px-4 text-sm font-bold"
    data-testid="button"
    type="button"
  />
</DocumentFragment>
`;

exports[`prop \`type\` renders with type="submit" 1`] = `
<DocumentFragment>
  <button
    class="bg-cta hover:bg-blue-400 text-white border-b-4 border-yellow-1 hover:border-blue-500 rounded py-2 px-4 text-sm font-bold"
    data-testid="button"
    type="submit"
  />
</DocumentFragment>
`;

exports[`renders 1`] = `
<DocumentFragment>
  <button
    class="bg-cta hover:bg-blue-400 text-white border-b-4 border-yellow-1 hover:border-blue-500 rounded py-2 px-4 text-sm font-bold"
    data-testid="button"
    type="button"
  />
</DocumentFragment>
`;
